File: common/src/test/java/dev/number6/slack/SlackClientIntegrationTest.java
package dev.number6.slack;

import dev.number6.slack.adaptor.SlackClientAdaptor;
import dev.number6.slack.port.SlackPort;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

public class SlackClientIntegrationTest {

    TestSlackClientComponent testee = DaggerTestSlackClientComponent.create();

    @Test
    void get() {
        SlackPort client = testee.slackPort();

        Optional<String> response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, String.class, new FakeLogger());
        assertThat(response.isPresent()).isTrue();
        assertThat(response.get()).isEqualTo("response");
    }

    @Test
    void post() {
        SlackPort client = testee.slackPort();

        Optional<String> response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, new FakeLogger(), "body");
        assertThat(response.isPresent()).isFalse();
    }

    @Test
    void postWithResponse() {
        SlackPort client = testee.slackPort();

        Optional<String> response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, "body", String.class, new FakeLogger());
        assertThat(response.isPresent()).isTrue();
        assertThat(response.get()).isEqualTo("response");
    }

    private static class FakeLogger implements LambdaLogger {

        @Override
        public void log(String message) {
        }

        @Override
        public void log(byte[] message) {
        }
    }
}

File: common/src/test/kotlin/dev/number6/slack/SlackClientIntegrationTest.kt
package dev.number6.slack

import assertk.assertThat
import assertk.assertions.isEqualTo
import assertk.assertions.isTrue
import assertk.assertions.isFalse
import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slack.adaptor.SlackClientAdaptor
import org.junit.jupiter.api.Test

class SlackClientIntegrationTest {
    var testee = DaggerTestSlackClientComponent.create()

    @Test
    fun get() {
        val client = testee.slackPort()
        val response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, String::class.java, FakeLogger())
        assertThat(response.isPresent()).isTrue()
        assertThat(response.get()).isEqualTo("response")
    }

    @Test
    fun post() {
        val client = testee.slackPort()
        val response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, FakeLogger(), "body")
        assertThat(response.isPresent()).isFalse()
    }

    @Test
    fun postWithResponse() {
        val client = testee.slackPort()
        val response = client.getSlackResponse(SlackClientAdaptor.CHANNEL_HISTORY_URL, "body", String::class.java, FakeLogger())
        assertThat(response.isPresent()).isTrue()
        assertThat(response.get()).isEqualTo("response")
    }

    private class FakeLogger : LambdaLogger {
        override fun log(message: String) {}
        override fun log(message: ByteArray) {}
    }
}